### PR TITLE
Fix: Make our app automatically aware of new Colony Contributors

### LINKED ISSUE #3632: [Payment -QA] The new follower isn't displayed in Simple Payment Recipient list
## Steps to reproduce

1. Join the New colony 
image

2. Ownew of colony make a simple payment, click on the Select member button, and enter "Junior" in the search field
image

## Expected behaviour

* All followers should appear in the recipient list to allow payments to any colony member.

## Actual behaviour

*  The new follower isn't displayed in Simple Payment Recipient list

### PR DESCRIPTION
**Pull request by rumzledz**
## Description

Issue #3622 is a bit misleading. It's not that the app doesn't ever show the new contributor. It's just that on `master`, you'd have to reload the app to make our app refetch our contributors list.

So I added a subscription logic to make the app automatically aware that a new contributor has been added to a Colony.

And the result is nothing short of spectacular ✨ 

app-aware-new-contributors

## Testing

1. Connect Leela's wallet
2. Create a new Colony using this script:

```console
node scripts/create-colony-url.js
```

3. Visit your new Colony
4. Stay on the Dashboard
5. Open an incognito tab
6. Connect Amy's wallet (Wallet 2)
7. Generate an invite code for the new Colony:

```gql
query MyQuery {
  getColonyByName(name: "your-new-colony's-name") { <<-- NOTE: Update this with your new Colony's name
    items {
      colonyMemberInviteCode
    }
  }
}
```

8. The result should be like this:

```gql
{
  "data": {
    "getColonyByName": {
      "items": [
        {
          "colonyMemberInviteCode": "cb488112-49af-4fd8-a666-5a3c71d35f1e"
        }
      ]
    }
  }
}
```

9. Copy the `colonyMemberInviteCode` field value i.e. `"cb488112-49af-4fd8-a666-5a3c71d35f1e"`
10. On the same Incognito tab, make sure that Amy's wallet is still connected
11. Visit http://localhost:9091/invite/{new-colony-name}/{colonyMemberInviteCode}
12. Proceed to join the Colony
13. Once you've successfully joined the Colony, switch back to the non-incognito window
14. **VERY IMPORTANT:** Please do not refresh the page
15. Open the Simple Payment form
16. Click the recipient field
17. Verify that your new user is visible on the list of Recipients
18. Visit the Members page
19. Verify that your new user is visible under the Followers tab 

Resolves #3632

### CODE REVIEW

File: src/context/MemberContext/MemberContextProvider.tsx [hunk 1]
```diff
@@ -119,7 +120,7 @@ const MemberContextProvider: FC<PropsWithChildren> = ({ children }) => {
     data: memberSearchData,
     loading,
     fetchMore,
-    refetch,
+    refetch: refetcColonyContributors,
```
[the remark this answers is not in the document]
Comment (rumzledz) [reply src/context/MemberContext/MemberContextProvider.tsx:123]: AH!

**Comment by rumzledz** [comment]
Those are irrelevant to the raised issue but sure, I'll fix it too 😂

**Comment by rumzledz** [comment]
@Nortsova  There! Irrelevant fix pushed 😂 🚀

File: src/hooks/useGetColoniesMembersCount.ts [hunk 2]
```diff
@@ -71,6 +75,27 @@ export const useGetColoniesMembersCount = (
...
+    useOnCreateColonyContributorSubscription();
+
+  const newColonyContributor =
+    newColonyContributorResult?.onCreateColonyContributor?.contributorAddress;
+
+  useEffect(() => {
+    let timeout: NodeJS.Timeout;
+
+    if (newColonyContributor) {
+      // It looks hacky, but we need the timeout to ensure that opensearch has been updated before we refetch.
```
Comment (iamsamgibbs) [inline src/hooks/useGetColoniesMembersCount.ts:88]: I know this fixes such a small edge case that it doesn't really matter too much, but is to `2000` timeout reliable for opensearch having updated? Does it reliably take less than 2 seconds?

**Review by davecreaser — APPROVED** [review]
Great solution @rumzledz! Subscriptions always feel so nice to use. I agree the timeout does look hacky, but it also makes sense 🤷‍♂️ 

Tested and working as expected, also in the members page:

Screenshot 2024-11-29 at 09 04 14
Screenshot 2024-11-29 at 09 04 28
Screenshot 2024-11-29 at 09 04 45

Nice work! 🎉

**Review by Nortsova — APPROVED** [review]
Awesome 🤩  

Thank you so much for updates:
image

**Pull request merged**